### ISSUE TITLE
Gates of Hell Bevavior

### ISSUE THREAD

**Issue by admkakaroto** [rathena/rathena#2956]
* **rAthena Hash**: ae69e506263f9b183c945f49b1d88c5fc2b73a6a
* **Client Date**: 2017-06-07
* **Server Mode**: Renewal
* **Description of Issue**: The HP damage bonus of the GoH skill is not working properly, the damage increases however very little if compared with the official server.
  * Result: The increase in damage is being negligible.
  * Expected Result: Considerable increase according to the player's HP.
  * Official Information: https://irowiki.org/wiki/Gates_of_Hell

```
Via Normal = (User_MaxHP − User_HP) + (User_SP × SP_Modifier) + (BaseLv × 10)
Via Combo = (User_MaxHP − User_HP) + (User_MaxSP × SP_Modifier) + (BaseLv × 40) 
```

Other problem:

- Ranged Atk modifier will not give any effects to this skill.

here in rathena is happening just the opposite.

**Comment by mrjnumber1** [comment]
your formula matches up with what i see in the zone.

skill info
```
SetSkill SR_GATEOFHELL //나찰파황격(SKID_SR_GATEOFHELL)
OnInit:
	SkillMaxLv		10
	SkillType		TOCHARACTER
	SkillFlag		SKFLAG_NODISPEL
	SkillPattern	SKILLPATT_DIRECTATTACK
	SkillRange					 1		2		3		3		4		4		5		5		6		7 // 스킬 사정거리
	SkillSP						11	   12	   13	   14	   15	   16	   17	   18	   19	   20 // SP소모량(0일때 사용하지 못하도록 임의설정)
	SkillCastFixedDelay          0	    0	    0	    0	    0	    0	    0	    0	    0	    0 // 스킬 캐스팅 고정 delay값
	SkillCastStatDelay		  1000	 1200	 1400	 1600	 1800	 2000	 2200	 2400	 2600	 2800 // 스킬 캐스팅 스텟에 영향받는 delay값
//	SkillSinglePostDelay	  5000	 5000	 5000	 5000	 5000	 5000	 5000	 5000	 5000	 5000 // 스킬사용후 해당스킬을 재사용하는데 필요한 시간
	SkillGlobalPostDelay       100	  200	  300	  400	  500	  600	  700	  800	  900	 1000 // 스킬사용후 전체스킬사용금지시간	
	SkillData1					12	   14	   16	   18	   20	   22	   24	   26      28	   30 // 데미지증폭 FACTOR 		
return
```

-code here https://paste2.org/BpkM1WIH

on ratehan it's doing ATK_ADD but it should directly be in the ratio based onthe code we see.

**Comment by admkakaroto** [comment]
Where is this code from?

**Comment by mrjnumber1** [comment]
it's a decompilation from episode 15.2 aegis zone

**Comment by aleos89** [comment]
Reported in #3259 and fixed in #5267.
